builtin/sparse-checkout.c

```diff
@@ -210,10 +210,13 @@ static void write_cone_to_file(FILE *fp, struct pattern_list *pl)
 	struct pattern_entry *entry;
 	struct hashmap_iter iter;
 	struct string_list sl = STRING_LIST_INIT_DUP;
+	struct strbuf parent_pattern = STRBUF_INIT;
 
 	hashmap_iter_init(&pl->parent_hashmap, &iter);
-	while ((entry = hashmap_iter_next(&iter)))
-		string_list_insert(&sl, entry->pattern);
+	while ((entry = hashmap_iter_next(&iter))) {
+		if (!hashmap_get(&pl->recursive_hashmap, entry, NULL))
+			string_list_insert(&sl, entry->pattern);
+	}
```

Comment:
Let me run a scenario by you and see if I am understanding thing correctly and it is handled.
If I have the following folders that I am setting `a` and `a/b/c`
`a` and `a/b/c` will be in the recursive hashmap right and parent hashmap will have `a`, and `a/b` right?
When we process `a/b` it will not be in the recursive hashmap and will still be added to the sparse-checkout file even though there is the `a` recursive entry?

Reply:
That's a good point. The solution I created is clearly not enough and I will need to revisit this by iterating up the parents. Good thing is that we can test this scenario with `deep` and `deep/deeper1/deepest`.